File: backend/bluffword-backend/src/main/java/org/bluffwordbackend/services/GameRoomBroadcaster.java
package org.bluffwordbackend.services;

import lombok.RequiredArgsConstructor;
import org.bluffwordbackend.models.GameRoomState;
import org.bluffwordbackend.redisDtos.PlayerDto;
import org.springframework.context.annotation.Configuration;
import org.springframework.messaging.simp.SimpMessagingTemplate;

import java.util.List;

@Configuration
@RequiredArgsConstructor
public class GameRoomBroadcaster {
    private final SimpMessagingTemplate messagingTemplate;

    public void broadcastPlayers(String roomCode, List<PlayerDto> players) {
        messagingTemplate.convertAndSend("/topic/room/" + roomCode +"/players", players);
    }

    public void broadcastGameRoomState(String roomCode, GameRoomState state) {
        messagingTemplate.convertAndSend("/topic/room/" + roomCode +"/state", state);
    }


}

File: backend_kotlin/src/main/kotlin/org/project/backend_kotlin/gameRoom/GameRoomBroadcaster.kt
package org.project.backend_kotlin.gameRoom

import org.project.backend_kotlin.redisModels.GameRoomState
import org.project.backend_kotlin.redisModels.Player
import org.springframework.messaging.simp.SimpMessagingTemplate
import org.springframework.stereotype.Service

@Service
class GameRoomBroadcaster(
    private val messagingTemplate: SimpMessagingTemplate
) {

    fun broadcastPlayers(roomCode: String, players: List<Player>) =
        messagingTemplate.convertAndSend("/topic/room/$roomCode/players", players)

    fun broadcastGameRoomState(roomCode: String, gameRoomState: GameRoomState) =
        messagingTemplate.convertAndSend("/topic/room/$roomCode/state", gameRoomState)



}